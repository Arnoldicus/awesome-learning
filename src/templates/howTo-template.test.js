import React from "react";
import renderer from 'react-test-renderer';
import HowToTemplate from "./howTo-template";

describe("HowTo template", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<HowToTemplate />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
