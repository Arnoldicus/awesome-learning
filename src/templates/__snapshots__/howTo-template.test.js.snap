// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowTo template matches the snapshot 1`] = `
Array [
  <StaticQuery
    render={[Function]}
  />,
  <div
    className="Layout Layout--fullBleed"
  >
    <div
      className="HowTo"
    >
      <div
        className="ContentSection"
      >
        <header
          className="ContentSection-title"
        >
          <h1>
            Gathering A Group
          </h1>
        </header>
        <div
          className="ContentSection-content HowTo-description ContentSection-content--alignLeft"
        >
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            Awesome Learning is best when run in a group setting. Your group can be co-located or remote. Here are some battle-tested ground rules.
          </p>
          <ul
            className="HowTo-list"
          >
            <li
              className="HowTo-listItem"
            >
              Limit the group size to a maximum of 6. Anything more can limit individual participation.
            </li>
            <li
              className="HowTo-listItem"
            >
              Strive to find a group of roughly similar skill levels. This isn't an exact science, but mixing highly experienced and in-experienced folks together can sometimes lead to being too bored or too challenged on a subject.
            </li>
            <li
              className="HowTo-listItem"
            >
              Set a time commitment early, and get buy-in. Shoot for an hour a week meeting with a consistent time and location.
            </li>
          </ul>
        </div>
      </div>
      <div
        className="ContentSection"
      >
        <header
          className="ContentSection-title"
        >
          <h1>
            Before Your Lesson
          </h1>
        </header>
        <div
          className="ContentSection-content HowTo-description ContentSection-content--alignLeft"
        >
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            Each lesson includes a video and reading materials. We strongly recommend working through both the videos and the reading materials
             
            <b>
              before
            </b>
             your lesson begins. If you are the lesson leader, please send out the link to the lesson to your group at least 24 hours in advance of when you will gather for the lesson exercises.
          </p>
          <ul
            className="HowTo-list"
          >
            <li
              className="HowTo-listItem"
            >
              <b>
                Good:
              </b>
               Watching the video before the lesson begins.
            </li>
            <li
              className="HowTo-listItem"
            >
              <b>
                Better:
              </b>
               Watching the video and reading the pre-read materials before the lesson begins.
            </li>
            <li
              className="HowTo-listItem"
            >
              <b>
                Best:
              </b>
               All of the above and taking the pre-read quiz to test your understanding.
            </li>
          </ul>
        </div>
      </div>
      <div
        className="ContentSection"
      >
        <header
          className="ContentSection-title"
        >
          <h1>
            During Your Lesson
          </h1>
        </header>
        <div
          className="ContentSection-content HowTo-description ContentSection-content--alignLeft"
        >
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            Check out this short video for an introduction to our exercise environment.
          </p>
          <iframe
            allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen={true}
            frameBorder="0"
            height="315"
            src="https://www.youtube.com/embed/e8_wPD6O3gA"
            width="560"
          />
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "16px",
              }
            }
          >
            <b>
              Rough lesson guide
            </b>
          </p>
          <ul
            className="HowTo-list"
          >
            <li
              className="HowTo-listItem"
            >
              Go over any questions the group has based on the pre-work materials.
            </li>
            <li
              className="HowTo-listItem"
            >
              Open and fork the exercise sets.
            </li>
            <li
              className="HowTo-listItem"
            >
              Read through all exercise descriptions and prompts out loud
            </li>
            <li
              className="HowTo-listItem"
            >
              Group program the exercises! Pass the keyboard for each exercise, discuss problems and solutions, etc. Above all, communicate!
            </li>
            <li
              className="HowTo-listItem"
            >
              Check the provided solutions if you get stuck, and don't be afraid to get derailed with discussions and research.
            </li>
          </ul>
        </div>
      </div>
      <div
        className="ContentSection"
      >
        <header
          className="ContentSection-title"
        >
          <h1>
            FAQs
          </h1>
        </header>
        <div
          className="ContentSection-content HowTo-description ContentSection-content--alignLeft"
        >
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            <b>
              Why should I Awesome Learn?
            </b>
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            Everyone in this industry deals with imposter syndrome. Technologies keep advancing while we feel like we're falling behind. Saying it out loud is the first step, changing it is the hard step.
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            <b>
              Who can Awesome Learn?
            </b>
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            Anyone at any level is welcome to start a group or jump in solo. We strongly believe learning can be done without an "expert" in the room.
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            <b>
              How does Awesome Learning work?
            </b>
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            Anyone can form a team. Once you gather a team, the team selects the "course" they want to learn first. Each course has pre-built hour-long lessons with learning materials and coding exercises included. We provide completed solutions and learning annotations to make sure you never get stuck.
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            <b>
              What will this do for my team?
            </b>
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            Awesome learning is a research-backed learning platform designed to help small teams build lasting, foundational skills. The platform is a series of videos, reading materials, and exercises in one hour intervals divided into subject-matter "courses". Reading quizzes keep learners engaged, surveys keep leaders up to date with how well team members are learning. TDD exercises provide a tight feedback loop.
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            <b>
              What is a Lesson?
            </b>
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            An Awesome Learning lesson is a series of videos, pre-reads, and exercises designed to teach you a specific subject matter within the scope of the course.
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            <b>
              What is a Course?
            </b>
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            An Awesome Learning course is a collection of lessons grouped under an umbrella theme. The specific goal of each lesson is to help you build a better mental model of the broader topic.
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            <b>
              How long is the time commitment for a single Course?
            </b>
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            It varies. Some of our courses contain 2 lessons, others contain 5. The recommended cadence is one lesson or per week. We don't reccommend a slower cadence than once per week. Our experience shows folks lose too much course context when more than a week passes in between lessons.
          </p>
        </div>
      </div>
    </div>
  </div>,
]
`;
